src/DotNetty.Codecs.Http/HttpServerCodec.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Codecs.Http
+{
+    using System.Collections.Generic;
+    using DotNetty.Buffers;
+    using DotNetty.Transport.Channels;
+
+    public class HttpServerCodec : CombinedChannelDuplexHandler<HttpRequestDecoder, HttpResponseEncoder>,
+        HttpServerUpgradeHandler.ISourceCodec
+    {
+        /** A queue that is used for correlating a request and a response. */
+        readonly Queue<HttpMethod> queue = new Queue<HttpMethod>();
```

Comment:
Java and .NET Queue types have different behavior of Poll/Dequeue (exception vs null). Is it possible this code (and Client one) for this code to encounter this corner case and will behavior be desired?